src/common/serviceRegistry.ts

```diff
@@ -0,0 +1,48 @@
+const defaultServiceRegistryInstance = {
+  taxonomyService: null,
+}
+
+export enum Services {
+  taxonomyService = 'taxonomyService',
+}
+
+const servicesByType = {
+  [Services.taxonomyService]: 'TaxonomyService',
+}
+
+let instance: any = null
+
+// registers
+const registerService = (type: Services) => {
+  if (!instance) {
+    instance = getInstance()
+  }
+  instance = {
+    ...instance,
+    [type]: instance[type] ?? servicesByType[type],
+  }
+  return instance
+}
+
+// getters
+const getService = (type: Services) => instance[type] || null
```

Comment:
we can implement a getXService:

const _getService = (type: Services) => () => instance[type] || null
const getXService = _getService(Services.type)

Reply:
Also, I can do the same with register methods.

@minotogna